daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala

```diff
@@ -779,24 +781,33 @@ object Main extends Logging {
           }
         }
       }
+      case "sax" => data
     }
   }
 
-  def getInfosetInputter(infosetType: String, anyRef: AnyRef): InfosetInputter = {
+  def getInfosetInputter(
+    infosetType: String,
+    anyRef: AnyRef,
+    processor: Option[DFDL.DataProcessor] = None,
+    outChannel: Option[DFDL.Output] = None): Either[InfosetInputter, DFDL.DaffodilInputContentHandler] = {
```

Comment:
Hmm, I don't see any benefit from declaring `x: Option[X] = None` at lines 791 and 792 only to call `x.get` afterwards at lines 809 and 810 below.  You're still going to get an exception, only it will be called `Predef.NoSuchElementException` instead of `java.lang.NullPointerException`.  If you don't want an exception, you would have to use getOrElse, a match expression, or flatten, flatMap, foreach, etc.

Reply:
The "= None" was intended as the default for the arguments so the nonSAX infosetTypes wouldn't need to pass in anything for the processor and outChannel params. But in both places I call it, we're passing in Some(_), so I removed the option and default.

I could change it so we pass in these 2 variables in a tuple via the anyRef, but I thought this way would be clearer since the anyRef is intended to represent inputterData